cmd/autoscaler/main.go

```diff
@@ -106,7 +106,7 @@ func main() {
 	ctx = logging.WithLogger(ctx, logger)
 
 	// statsCh is the main communication channel between the stats server and multiscaler.
-	statsCh := make(chan *autoscaler.StatMessage, statsBufferLen)
+	statsCh := make(chan autoscaler.StatMessage, statsBufferLen)
```

Comment:
So I considered this change before, but here we use a buffered channel, which means each object will be copied at least twice (in case of unbuffered channel it's usually 0 or stack to stack transfer once).
So it might be fine to keep as is. But may be we need some measurements. :)